### ISSUE TITLE
Add support to use `--fix` with `checkAgainstRule`

### ISSUE THREAD

**Issue by aaronccasanova** [stylelint/stylelint#6465]
### What is the problem you're trying to solve?

The `--fix` option does not work with the `checkAgainstRule` utility:

```js
stylelint.utils.checkAgainstRule(
  {
    ruleName: 'length-zero-no-unit'
    ruleSettings: true,
    root,
  },
  (warning) => { /* Don't report if the problem can be automatically fixed */ },
)
```

### What solution would you like to see?

I would like to be able to pass through the `context.fix` option to `checkAgainstRule`:

```js
module.exports = createPlugin(
  'custom/length-zero-no-unit',
  (primary, secondary, context) => (root) => {
    stylelint.utils.checkAgainstRule(
      {
        ruleName: 'length-zero-no-unit'
        ruleSettings: true,
        fix: context.fix,
        root,
      },
      (warning) => { /* Don't report if the problem can be automatically fixed */ },
    )
  },
})
```

My team currently has this behavior patched with the following updates to `checkAgainstRule`:
https://github.com/stylelint/stylelint/blob/64b022a052eba7d5a6358781d19cdd3f84a2a0b3/lib/utils/checkAgainstRule.js#L47
```diff
- rule(settings[0], /** @type {O} */ (settings[1]), {})(options.root, tmpPostcssResult);
+ rule(settings[0], /** @type {O} */ (settings[1]), {fix: options.fix})(options.root, tmpPostcssResult);
```

**Comment by ybiquitous** [comment]
@aaronccasanova Thanks for the proposal. I agree with passing the `fix` option to `checkAgainstRule()`. Furthermore, giving a `context` object seems to make sense instead of `fix`. What do you think?

**Comment by aaronccasanova** [comment]
Great idea, passing through the [PluginContext](https://github.com/stylelint/stylelint/blob/d91bb5b779ce6ca69058549d58195a9e82b7c8d1/types/stylelint/index.d.ts#L151-L154) is definitely more future facing 👍

**Comment by ybiquitous** [comment]
@aaronccasanova Thanks for the feedback. Let's add the `context` parameter to `checkAgainstRule`.

I've labeled the issue as **ready to implement**. Please consider [contributing](https://stylelint.io/contributing) if you have time.
